public/app/core/components/RolePicker/RolePicker.tsx

```diff
@@ -65,7 +67,10 @@ export const RolePicker = ({
       vertical = offsetVertical;
     }
 
-    if (window.innerWidth - right < ROLE_PICKER_WIDTH) {
+    if (
+      window.innerWidth - right < currentRolePickerWidth &&
+      currentRolePickerWidth < 2 * ROLE_PICKER_SUBMENU_MIN_WIDTH
```

Comment:
Not sure why we need to compare to 2x of submenu width? I see some cases where there's enough space on the right for the submenu, but it gets rendered on the left instead.

Reply:
Without this the submenu appears on the right of the RolePicker when the RolePicker is wider than its submenu 2 * min-width and I think it looks strange:
![image](https://user-images.githubusercontent.com/1585112/186373545-01eadfcd-77d6-47c3-9ef1-04c2146ed58c.png)

So I decided to add this check to improve the placement of the submenu (it appears on the left of the RolePicker input box) in case the RolePicker is wide.

Reply:
> That's indeed strange, is that on org/teams/new page?

Nope, the screenshot I sent was just an example (I changed the width to 1000px to show why the change is beneficial). Currently RolePicker looks good on every page (because the max-width is 600px as you mentioned). I just wanted to include this improvement to make the RolePicker look good when it is wider than 620px. Especially, I wanted to make that calculation universal for the RolePicker to work correctly regardless of its width.

Reply:
Would `if (window.innerWidth - right < currentRolePickerWidth && currentRolePickerWidth < ROLE_PICKER_SUBMENU_MIN_WIDTH) {` work then? It looks fine with an extra long form (which should never happen in practice tbh) and it properly adjusts inside the lists.

Reply:
I don't think so (screenshot attached). Essentially, the second part of the expression calculates whether there is enough place under the RolePicker to show/fit both (the role picker menu and its sub menu) aligned to the left edge of the input. Otherwise, it aligns the role picker menu to the right.

![image](https://user-images.githubusercontent.com/1585112/186474189-3a3904f5-c772-4f53-a03e-cb14ea20db41.png)

Reply:
Ah I see, thanks for clarification! Maybe also a good idea to add a brief comment for that condition in the code.

Reply:
Done